File: simplified-opds-core/src/main/java/org/nypl/simplified/opds/core/OPDSFeedTransportException.java
package org.nypl.simplified.opds.core;

/**
 * The type of exceptions raised by feed transports.
 */

public abstract class OPDSFeedTransportException extends Exception
{
  /**
   * Construct an exception.
   *
   * @param message The message
   * @param cause   The cause
   */

  public OPDSFeedTransportException(
    final String message,
    final Throwable cause)
  {
    super(message, cause);
  }

  /**
   * Construct an exception.
   *
   * @param message The message
   */

  public OPDSFeedTransportException(
    final String message)
  {
    super(message);
  }
}

File: simplified-opds-core/src/main/java/org/nypl/simplified/opds/core/OPDSFeedTransportException.kt
package org.nypl.simplified.opds.core

/**
 * The type of exceptions raised by feed transports.
 */

abstract class OPDSFeedTransportException : Exception {

  /**
   * Construct an exception.
   *
   * @param message The message
   * @param cause   The cause
   */

  constructor(
    message: String,
    cause: Throwable
  ) : super(message, cause)

  /**
   * Construct an exception.
   *
   * @param message The message
   */

  constructor(message: String) : super(message)
}
